pkg/controller/workloadspread/workloadspread_event_handler.go

```diff
@@ -0,0 +1,216 @@
+/*
+Copyright 2021 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workloadspread
+
+import (
+	"context"
+	"encoding/json"
+
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+	kubecontroller "k8s.io/kubernetes/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	appsalphav1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+	wsutil "github.com/openkruise/kruise/pkg/util/workloadspread"
+)
+
+type EventAction string
+
+const (
+	CreateEventAction EventAction = "create"
+	UpdateEventAction EventAction = "update"
+	DeleteEventAction EventAction = "delete"
+)
+
+var _ handler.EventHandler = &podEventHandler{}
+
+type podEventHandler struct{}
+
+func (p *podEventHandler) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	p.handlePod(q, evt.Object, CreateEventAction)
+}
+
+func (p *podEventHandler) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	oldPod := evt.ObjectOld.(*corev1.Pod)
+	newPod := evt.ObjectNew.(*corev1.Pod)
+
+	if kubecontroller.IsPodActive(oldPod) && !kubecontroller.IsPodActive(newPod) {
```

Comment:
Don't really understand why need to check this?

Reply:
The WorkloadSpread controller counts the active Pods to maintain the real missingReplicas of subset. Therefore, it only care about the update event that the Pod has terminated its lifecycle. For example, `DeletionTimestamp` from nil to not nil, status.Phase become `succeed` or `failed`.